app.rb

```diff
@@ -0,0 +1,188 @@
+        if arg.is_a?(Class)
+          return false unless j.is_a?(arg)
+        elsif arg.class == Regexp
+          return false unless arg.match(j)
+        else
+          return false unless arg == j
+        end
+      end
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    each do |j|
+      if block_given?
+        return true if yield j
+      end
+      if arg.nil? && !block_given?
+        return true if j
+      end
+      unless arg.nil?
+        if arg.is_a?(Class)
+          return true if j.is_a?(arg)
+        elsif arg.class == Regexp
+          return true if arg.match(j)
+        else
+          return true unless arg != j
+        end
+      end
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    each do |j|
+      if block_given?
+        return false if yield j
+      end
+      if arg.nil? && !block_given?
+        return false if j
+      end
+      if arg.class == Regexp
+        return false if arg.match(j)
+      end
+      unless arg.nil? && !arg.is_a?(Class)
+        return false if arg == j
+      end
+      if arg.is_a? Class
+        return false if j.is_a?(arg)
+      end
+    end
+    true
+  end
+
+  def my_count(counted = nil)
+    new_array = is_a?(Range) ? to_a : self
+    return new_array.length if counted.nil? && !block_given?
+
+    counting = 0
+    if block_given?
+      new_array.my_each { |j| counting += 1 if yield j }
+    end
+    unless counted.nil?
+      new_array.my_each { |j| counting += 1 if j == counted }
+    end
+    counting
+  end
+
+  def my_map(proc = nil)
+    return enum_for unless block_given?
+
+    arr = []
+    each do |i|
+      if !proc.nil?
+        arr.push(proc.call(i))
+      else
+        arr.push(yield i)
+      end
+    end
+    arr
+  end
+
+  def my_inject(*arg)
+    new_array = is_a?(Range) ? to_a : self
+    product = 0
+    long_word = ''
+    i = 0
+    temp = new_array[0]
+    while i < new_array.length - 1
+      long_word = yield temp, new_array[i + 1] if new_array[i].is_a?(String)
+      i += 1
+    end
+    return long_word if long_word != ''
+
+    if !block_given?
+      if arg[0].class == Symbol
+        new_array.my_each { |j| product = product.send(arg[0], j) }
+        if product.zero?
+          product = 1
+          new_array.my_each { |j| product = product.send(arg[0], j) }
+        end
+      elsif arg[0].class != Symbol && !arg[1].nil?
+        new_array.my_each { |j| product = product.send(arg[1], j) }
+        if product.zero?
+          product = 1
+          new_array.my_each { |j| product = product.send(arg[1], j) }
+        end
+        product = product.send(arg[1], arg[0])
+      end
+    else
+      new_array.my_each { |j| product = yield product, j }
+      if product.zero?
+        product = 1
+        new_array.my_each { |j| product = yield product, j }
+      end
+      product = yield product, arg[0] unless arg[0].nil?
+    end
+    raise LocalJumpError unless block_given?
+
+    product
+  end
+end
```

Comment:
We have the error: 👇  in the following tests.
```
     LocalJumpError:
       LocalJumpError
```
- [ ] When a block is given when self is a range behaves like #each_with_index calls the given block once for each element in self.
- [ ] When a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method.
- [ ] When a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method:range
- [ ] When a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method
- [ ] When a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method::range

- If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email. :eyes:
- When you have made the specified changes, please submit another code review. Thank you!
Happy coding :smile:
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)
---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***